client/utils/api.js

```diff
@@ -0,0 +1,28 @@
+import axios from 'axios';
+
+/**
+ * ApiCall - The base for all API call
+ * @param {object} data
+ * @param {string} type
+ * @param {string} url
+ * @returns {object}
+ */
+export default function apiCall(url, type, data,) {
+  return new Promise((resolve, reject) => {
```

Comment:
Well done @chibujax but I feel this implementation could be better if we use async-await to resolve the promise so that we can return the actual response data.

Reply:
Thank you so much for spotting those errors. For this returning the actual response payload, yes it does that, look at line 22

Reply:
@chibujax Okay... seen. :+1: